libkbfs/block_tree.go

```diff
@@ -403,11 +409,59 @@ func (bt *blockTree) processGetBlocksTask(ctx context.Context,
 		} else {
 			results <- getBlocksForOffsetRangeResult{
 				pathFromRoot: childPath,
+				firstBlock:   childIsFirstBlock,
 			}
 		}
+		childIsFirstBlock = false
 	}
 }
 
+func checkForHolesAndTruncate(
+	pathsFromRoot [][]parentBlockAndChildIndex) [][]parentBlockAndChildIndex {
+	var prevPath []parentBlockAndChildIndex
+	for pathIdx, path := range pathsFromRoot {
+		// Each path after the first must immediately follow the preceding path.
+		pathIsCorrect := true
+		if pathIdx != 0 {
```

Comment:
Heh, I did it this way to avoid having multiple `return` locations. But I can do it that way instead.

Reply:
Multiple returns aren't a huge deal in golang, compared to other languages, because of the nice `defer` statements.  So I tend to prefer to optimize for indentation, but your mileage might vary.